### PR TITLE
[HGCal trigger] Updates in Module sums and Towers

### PR DESCRIPTION
**Pull request by jbsauvan**
#### PR description:
- Use all Trigger Cells to build Module Sums
- Use Module Sums from all TCs to build Towers. This change is required to reflect the hardware/firmware.
- Add Tower building in the HFNose

Associated internal PRs and reviews:
- PFCal-dev#463
- PFCal-dev#467
- PFCal-dev#477

Related presentation in HGCAL TPG meetings:
- https://indico.cern.ch/event/980323/#2-updates-on-jets-with-towers

#### PR validation:
Private tests in `L1Trigger/L1THGCalUtilities/test`:
```
cmsRun testHGCalL1T_RelValV11_cfg.py
cmsRun testHGCalL1T_multialgo_V11_cfg.py
```
Tested D49, D60 and D68 workflows
```
runTheMatrix.py -w upgrade -l 23234.0 --maxSteps=2 
runTheMatrix.py -w upgrade -l 28234.0 --maxSteps=2 
runTheMatrix.py -w upgrade -l 31434.0 --maxSteps=2
```
---
@mariadalfonso @echapon FYI

### CODE REVIEW

**Comment by srimanob** [comment]
Please test

**Comment by jbsauvan** [comment]
I just saw it didn't pass the checks (it seems that I didn't receive a notification from `cmsbuild`). 
I'll fix the errors.

**Comment by srimanob** [comment]
Please abort

**Comment by jbsauvan** [comment]
I am fixing the missing header include.
I see there a also several warnings regarding deprecated `EventSetupRecord::get(ESHandle&)`, shall I fix them in this PR?

**Comment by qliphy** [comment]
> I am fixing the missing header include.
> I see there also several warnings regarding deprecated `EventSetupRecord::get(ESHandle&)`, shall I fix them in this PR?

Not necessary, the warnings are on the codes not touched by you.

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/L1THGCal/python/customTriggerSums.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import FWCore.ParameterSet.Config as cms
+import math
```
Comment (srimanob) [inline L1Trigger/L1THGCal/python/customTriggerSums.py:2]: Why do we need math here? 
Thx.

Comment (jbsauvan) [reply L1Trigger/L1THGCal/python/customTriggerSums.py:2]: Indeed it doesn't seem we need it.
I'll remove it.
Thanks!

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
@jbsauvan 
One small comment, why you update the test configs from D35 to D49, not newer geometry? D49 will be dropped at any moment. However, this is not a showstopper, as it is not my task to check the test config you prefer. Just curious. Thx.

**Comment by jbsauvan** [comment]
Hi @srimanob 
This is because we are using HLT TDR samples for our tests.
But indeed once D49 is dropped, we'll have to switch to other test samples.

**Comment by cecilecaillol** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**